File: backend/src/modules/report/report.controller.js
import { getProjectReport } from './report.repository.js';

export const getReport = async (req, res) => {
    try {
        const { id } = req.params;
        const id_user = req.user.id;
        const report = await getProjectReport(id, id_user);
        if (!report) return res.status(404).json({ error: 'Proyecto no encontrado.' });
        res.status(200).json(report);
    } catch (error) {
        res.status(500).json({ error: error.message });
    }
};

File: QALearnBackend/Controllers/ReportController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using QALearnAPI.Repositories;

namespace QALearnAPI.Controllers;

[ApiController, Route("api/projects/{id:guid}/report"), Authorize]
public class ReportController(ReportRepository repo) : ControllerBase
{
    [HttpGet]
    public async Task<IActionResult> GetReport(Guid id)
    {
        var userId = Guid.Parse(User.FindFirst("id")!.Value);
        var report = await repo.GetProjectReportAsync(id, userId);
        if (report == null) return NotFound(new { error = "Proyecto no encontrado." });
        return Ok(report);
    }
}
